src/schema/search.ts

```diff
@@ -66,26 +63,85 @@ const hydrateSearchResult = (result, root) => {
 
   const loader = loaderMapping[result.label]
 
-  return loader(result.id).then(object =>
-    assign(object, { __typename: result.label })
-  )
+  return loader(result.id).then(object => {
+    return {
+      ...object,
+      __typename: result.label,
+    }
+  })
 }
 
-const ArtistConnection = connectionWithCursorInfo(Searchable)
+const SearchConnection = connectionWithCursorInfo(Searchable)
 
-export const Search = {
-  type: ArtistConnection,
+export const Search: GraphQLFieldConfig<any, any, any> = {
+  type: SearchConnection,
   description: "Global search",
   args: searchArgs,
-  resolve: (root, options, _request) => {
-    return root.searchLoader(options).then(({ body }) => {
+  resolve: (source, args, _request, info) => {
+    const pageOptions = convertConnectionArgsToGravityArgs(args)
+
+    let shouldHydrate = false
+
+    visit(info.fieldNodes[0], {
+      Field(node) {
+        if (node.name.value === "node") {
+          visit(node, {
+            InlineFragment(node) {
+              if (
+                node.typeCondition &&
+                node.typeCondition.name.value !== SearchableItem.name
+              ) {
+                shouldHydrate = true
+              }
+            },
+            FragmentSpread(_node) {
+              throw new Error(
+                "Named fragment spreads are currently unsupported for search."
+              )
+            },
+          })
```

Comment:
We can probably return the BREAK constant here to stop further recursion of the AST